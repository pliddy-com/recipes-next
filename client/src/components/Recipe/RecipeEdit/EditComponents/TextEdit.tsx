import InputAdornment from '@mui/material/InputAdornment';
import TextField from '@mui/material/TextField';
import { ChangeEvent, ReactElement } from 'react';

interface ITextEdit {
  className?: string;
  disabled?: boolean;
  endAdornment?: ReactElement | string;
  id: string;
  label: string;
  onChange?: (e: ChangeEvent<HTMLTextAreaElement | HTMLInputElement>) => void;
  value: string | number;
}
const TextEdit = ({
  className,
  disabled = false,
  endAdornment,
  id,
  label,
  onChange,
  value
}: ITextEdit) => {
  const handleUpdate = (
    e: ChangeEvent<HTMLTextAreaElement | HTMLInputElement>
  ) => {
    e.preventDefault();
    onChange && onChange(e);
  };

  return (
    <TextField
      className={`field${className ? ` ${className}` : ''}`}
      disabled={disabled}
      InputProps={{
        inputProps: { 'data-testid': id },
        ...(endAdornment
          ? {
              endAdornment: (
                <InputAdornment position="end">{endAdornment}</InputAdornment>
              )
            }
          : {})
      }}
      id={id}
      label={label}
      name={id}
      onChange={onChange ? (e) => handleUpdate(e) : undefined}
      size="small"
      value={value}
      variant="outlined"
    />
  );
};

export default TextEdit;
