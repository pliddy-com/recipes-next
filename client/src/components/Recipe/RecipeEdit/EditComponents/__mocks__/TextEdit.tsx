import { ChangeEvent } from 'react';

jest.createMockFromModule('../TextEdit');

export const TextEdit = jest.fn(
  ({
    label,
    onChange,
    value
  }: {
    label: string;
    onChange: (e: ChangeEvent<HTMLTextAreaElement | HTMLInputElement>) => void;
    value: string | number;
  }) => {
    return (
      <div data-testid={`mock-text-field-${label}`}>
        <label>{label}</label>
        <input value={value} onChange={onChange} />
      </div>
    );
  }
);

export default TextEdit;
