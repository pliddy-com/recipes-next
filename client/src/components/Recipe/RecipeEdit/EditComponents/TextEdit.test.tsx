import '@testing-library/jest-dom';
import { fireEvent, render } from '@testing-library/react';

import TextEdit from './TextEdit';

describe('TextEdit', () => {
  describe('when there is content', () => {
    it('renders the component', () => {
      const testValue = 'test value';

      const props = {
        className: 'bold',
        id: 'test-id',
        label: 'Test ID',
        onChange: jest.fn(),
        value: testValue
      };

      const targetValue = 'target value';

      const { asFragment, queryByLabelText } = render(<TextEdit {...props} />);
      expect(asFragment()).toMatchSnapshot();

      const testInput = queryByLabelText(props.label);

      testInput &&
        fireEvent.change(testInput, {
          target: { value: targetValue }
        });
    });
  });

  describe('with optional props', () => {
    it('renders the component', () => {
      const props = {
        disabled: false,
        endAdornment: 'Max',
        id: 'test-id',
        label: 'Test ID',
        value: 'test value'
      };
      const { asFragment } = render(<TextEdit {...props} />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
