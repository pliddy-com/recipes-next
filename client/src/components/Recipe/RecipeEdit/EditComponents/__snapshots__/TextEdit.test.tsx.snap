// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextEdit when there is content renders the component 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiTextField-root field bold css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
      for="test-id"
      id="test-id-label"
    >
      Test ID
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
        data-testid="test-id"
        id="test-id"
        name="test-id"
        type="text"
        value="test value"
      />
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-14lo706"
        >
          <span>
            Test ID
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TextEdit with optional props renders the component 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiTextField-root field css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
      for="test-id"
      id="test-id-label"
    >
      Test ID
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedEnd css-19qh8xo-MuiInputBase-input-MuiOutlinedInput-input"
        data-testid="test-id"
        id="test-id"
        name="test-id"
        type="text"
        value="test value"
      />
      <div
        class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeSmall css-1laqsz7-MuiInputAdornment-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1pnmrwp-MuiTypography-root"
        >
          Max
        </p>
      </div>
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-14lo706"
        >
          <span>
            Test ID
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</DocumentFragment>
`;
